### PR TITLE
stdlib: remove deprecated floating point APIs

### PR DESCRIPTION
**Pull request by gribozavr**
Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by gribozavr** [comment]
@stephentyrone Could you review for the preview-1 branch, please?

**Comment by gribozavr** [comment]
@swift-ci Please test

File: stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb [hunk 1]
```diff
@@ -136,9 +136,9 @@ public struct CGFloat {
     return CGFloat(NativeType.signalingNaN)
   }
   
-  @available(*, deprecated, renamed: "nan")
+  @available(*, unavailable, renamed: "nan")
   public static var quietNaN: CGFloat {
-    return CGFloat(NativeType.nan)
+    fatalError("unavailable")
```
Comment (stephentyrone) [inline stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb]: Should these be `Builtin.unreachable()`?

**Comment by stephentyrone** [comment]
LGTM, modulo question about `Builtin.unreachable( )`, and check failures (which look like they may be noise).

File: stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb (hunk 1 above)
Comment (gribozavr) [inline stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb]: Unfortunately we can't use `Builtin` outside of the core library without changing the build system.

**Comment by gribozavr** [comment]
Thanks @stephentyrone!

**Comment by gribozavr** [comment]
The CI issues look unrelated to me, let's try again.

@swift-ci Please test and merge

File: stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb (hunk 1 above)
Comment (jckarter) [inline stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb]: There's a transparent `_unreachable` wrapper in the stdlib, isn't there?

**Pull request merged**